src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/UseValidPlatformStringTest.cs

```diff
@@ -376,6 +376,31 @@ public class Test
                 VerifyCS.Diagnostic(UseValidPlatformString.NoVersion).WithLocation(4).WithArguments("4.1", "Linux"));
         }
 
+        [Fact]
+        public async Task PlatformOSXIsAliasForMacOS()
+        {
+            var csSource = @"
+using System.Runtime.Versioning;
+
+public class Test
+{
+    [{|#0:SupportedOSPlatform(""MacOS1.2.3.4"")|}] // Version '1.2.3.4' is not valid for platform 'MacOS'. Use a version with 2-3 parts for this platform.
+    public void SupportedOSPlatformMac4PartsInvalid() { }
+
+    [SupportedOSPlatform(""MacOS1.2"")]
+    public void SupportedMacOs2PartValid() { }
+
+    [{|#1:SupportedOSPlatform(""OSX1.2.3.4"")|}] // Version '1.2.3.4' is not valid for platform 'OSX'. Use a version with 2-3 parts for this platform.
+    public void SupportedOSPlatformOSX4PartsInvalid() { }
+
+    [SupportedOSPlatform(""Osx1.2"")]
+    public void SupportedOsx2PartValid() { }
+}";
```

Comment:
What would happen if you add a method topped with two `[SupportedOSPlatform]` attributes, one with "OSX" and another one with "MacOS"? Maybe we should have a test for that edge case too.

Reply:
Nothing should happen, the analyzer should not distinguish them as same nor should require to remove one of them.

I can add a test to show nothing happening